examples/python/sample_mlc_chat.py

```diff
@@ -25,3 +25,19 @@
 
 # Reset the chat module by
 # cm.reset_chat()
+
+# Stream using an Iterator
+from threading import Thread
+
+stream = StreamIterator(callback_interval=2)
+generation_thread = Thread(
+    target=cm.generate,
+    kwargs={"prompt": "What is the meaning of life?", "progress_callback": stream},
+)
+generation_thread.start()
+
+output = ""
+for delta_message in stream:
+    output += delta_message
+
+generation_thread.join()
```

Comment:
Thanks @varshith15 for contribution! I am wondering if we can move the example usage to a separate example file? This is mainly to keep the both examples minimal.

Reply:
yes, can do that
but I feel like streamiterator is just a continuation of streamtostout so it should ideally be part of the same file?
but if you recommend otherwise ill move to a different file.
Also should I add it to the docs? @MasterJH5574
let me know, ill push the changes asap

Reply:
Either way works for me, and I slightly prefer having a separate example with a more accurate name. What do you guys think?

Reply:
@junrushao  @MasterJH5574  please review, I've moved the example to a seperate file and also added the code snippet to the docs